.circleci/regenerate.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/env python3
+
+# regenrates config.yml based on config.yml.in
+
+from copy import deepcopy
+import os.path
+
+import jinja2
+import yaml
+from jinja2 import select_autoescape
+
+WORKFLOWS_JOBS_PR = {"filters": {"branches": {"ignore": ["master"]}}}
+
+WORKFLOWS_JOBS_MASTER = {
+    "context": "org-member",
+    "filters": {"branches": {"only": ["master"]}},
+}
+
+
+def indent(indentation, data_list):
+    return ("\n" + " " * indentation).join(
+        yaml.dump(data_list, default_flow_style=False).splitlines()
+    )
+
+
+def jobs(pr_or_master, num_workers=20, indentation=2):
+    w = {}
+    needs_gpu_nvidia_small_multi = [3, 9, 12, 13]
```

Comment:
Do we know why 3, 9, 12, and 13 need nvidia small multi? Same as the second line

Reply:
no clue, i took them from the original config.yml, i assume its because these shards end up running tutorials that need those machines but i cant find info about which tutorials specifically need them

when i do sharding w/ some sort of meta data ill probably figure it out